### ISSUE TITLE
boto_vpc.route_table_present error with nat_gateway_subnet_name

### ISSUE THREAD

**Issue by twellspring** [saltstack/salt#38791]
### Description of Issue/Question
Error when create a routing table and associate it with an already existing NAT Gateway using the nat_gateway_subnet_name option. 

### Setup
This is the failing state. 

```
aws_vpc_vpcProdWest_create_routing_table_privateC:
  boto_vpc.route_table_present:
    - name: privateC
    - vpc_name: vpcProdWest
    - profile: {'region': 'us-east-2', 'keyid': 'XXXXXXXXXX', 'key': 'XxXXXXXXXXXXXXXXXXXXXXXXXXXXX'}
    - routes:
      - nat_gateway_subnet_name: 'subWebC'
        destination_cidr_block: '0.0.0.0/0'
```

Subnet subWebC exists and has a Nat Gateway

```
salt   --verbose --force-color  'local-core-api-prod-1' boto_vpc.describe_nat_gateways  subnet_name=subWebC 'profile={'region': 'us-east-2', 'keyid': 'xxxxxxxxxx', 'key': 'xxxxxxxxxxxxxxxxxxxx'}'

local-core-api-prod-1:
    |_
      ----------
      CreateTime:
          ¸20170114T00:40:54.132000
      NatGatewayAddresses:
          |_
            ----------
            AllocationId:
                eipalloc-84d913ed
            NetworkInterfaceId:
                eni-0a50e647
            PrivateIp:
                10.20.3.15
            PublicIp:
                52.14.35.129
      NatGatewayId:
          nat-083136f95d5bc9c11
      State:
          available
      SubnetId:
          subnet-466e790c
      VpcId:
          vpc-4d65b624
```

But the state fails
```local-core-api-prod-1:
----------
          ID: aws_vpc_vpcProdWest_create_routing_table_privateC
    Function: boto_vpc.route_table_present
        Name: privateC
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1744, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1702, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/dist-packages/salt/states/boto_vpc.py", line 952, in route_table_present
                  region=region, key=key, keyid=keyid, profile=profile)
                File "/usr/lib/python2.7/dist-packages/salt/states/boto_vpc.py", line 1123, in _routes_present
                  keyid=keyid, profile=profile, **r)
                File "/usr/lib/python2.7/dist-packages/salt/modules/boto_vpc.py", line 2349, in create_route
                  NatGatewayId=nat_gateway_id)
                File "/usr/local/lib/python2.7/dist-packages/botocore/client.py", line 251, in _api_call
                  return self._make_api_call(operation_name, kwargs)
                File "/usr/local/lib/python2.7/dist-packages/botocore/client.py", line 537, in _make_api_call
                  raise ClientError(parsed_response, operation_name)
              ClientError: An error occurred (InvalidParameterValue) when calling the CreateRoute operation: route table rtb-7fa97116 and interface interface-0a50e647 belong to different networks
     Started: 19:52:18.805296
    Duration: 1945.241 ms
     Changes:

Summary for local-core-api-prod-1
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   1.945 s
```

### Steps to Reproduce Issue
minion debug log attached. 
Can reproduce this problem with an existing or a new routing table.  

### Versions Report
Master is MacOS  Minion is Ubuntu 14.04 with the same salt version.

Minion has these boto libraries
```
root@local-core-api-prod-1:~# pip list | grep boto
DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.
boto (2.45.0)
boto3 (1.4.3)
botocore (1.4.92)
```

**MINION**
```
root@local-core-api-prod-1:~# salt-minion --versions-report
Salt Version:
           Salt: 2016.11.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.0
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.6 (default, Oct 26 2016, 20:30:19)
   python-gnupg: 0.3.6
         PyYAML: 3.10
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-96-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

**MASTER**
```
± |master S:9 U:2 ?:5 ✗| → salt --versions-report
Salt Version:
           Salt: 2016.11.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.10 (default, Oct 23 2015, 19:19:21)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4.2
            ZMQ: 4.2.0

System Versions:
           dist:
        machine: x86_64
        release: 15.6.0
         system: Darwin
        version: 10.11.6 x86_64
```
[minion.debug.txt](https://github.com/saltstack/salt/files/712334/minion.debug.txt)

**Comment by Ch3LL** [comment]
@twellspring can you confirm that you can manually add this nat gateway subnet as a route table.

I see this error: `route table rtb-7fa97116 and interface interface-0a50e647 belong to different networks` in the error output which pushes me to believe the nat gateway and the route aren't in the same network. Although this is not my strong area. But if you could confirm you can add it manually to confirm its on salts end and not an issue with aws that would be awewsome thanks

**Comment by twellspring** [comment]
The NAT Gateway can be added to the route either manually or via salt by using `nat_gateway_id`.   

Here is a salt state that works using nat_gateway_id.

``` 
aws_vpc_vpcProdEast5_create_routing_table_privateC:
  boto_vpc.route_table_present:
    - name: privateC
    - vpc_name: vpcProdEast
    - profile: {'region': 'us-east-2', 'keyid': 'XXXXXXXXXX', 'key': 'XXXXXXXXXXXXXXXXXXXXX'}
    - routes:
      - destination_cidr_block: '0.0.0.0/0'
        nat_gateway_id: 'nat-XXXXXXXXXXXXXXXXXXX'
```

**Comment by twellspring** [comment]
I am now able to use the `nat_gateway_subnet_name` parameter.   
When I tried to do some more testing of this issue, I found that it worked.   What I have discovered is the boto states/modules do not correctly respect the VPC of an item.  An example of this is using the boto_vpc.route_table_present with a routing table that already exists in another VPC will complete successful with no changes.

I believe what was happening above is I had another VPC and the boto_vpc.route_table_present state was trying to add the nat gateway for this subnet to the routing table of the other VPC. 

I will put in a new issue for this problem.
